.github/workflows/deploy.yml

```diff
@@ -26,21 +46,79 @@ jobs:
         server-id: openconext-snapshots
         server-username: MAVEN_USERNAME
         server-password: MAVEN_PASSWORD
+      if: ( endsWith(steps.versioncheck.outputs.version, '-SNAPSHOT')) 
+    
+    - name: Set up JDK 17 for releases
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: 'maven'
+        server-id: openconext-releases
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_PASSWORD
+      if: ${{!( endsWith(steps.versioncheck.outputs.version, '-SNAPSHOT')) }}
+
     - name: Set up cache
```

Comment:
Should the cache be set up before running maven for the first time (versioncheck)?